### PR TITLE
feat(schema-engine): disabled implicit transaction for batch statements starting from CockroachDB v22.2

### PR DESCRIPTION
**Pull request by jkomyno**
This PR closes https://github.com/prisma/prisma/issues/20851.

It performs the same logic previously applied to CockroachDB v22.2 only, to any CockroachDB version equal or more recent than v22.2.

It also adds the CockroachDB-specific logic to parse the semantic versions. This implies that, at some point, we could disable the `enable_implicit_transaction_for_batch_statements` once a new major/minor version is reached.

### CODE REVIEW

File: schema-engine/sql-migration-tests/tests/migrations/cockroachdb.rs [hunk 1]
```diff
@@ -451,6 +451,72 @@ fn connecting_to_a_cockroachdb_database_with_the_postgresql_connector_fails(_api
...
+
+                */
+                -- AlterTable
+                ALTER TABLE "Cat" DROP COLUMN "tag";
+                ALTER TABLE "Cat" ADD COLUMN     "tag" BOOL NOT NULL;
+            "#]];
+
+            migration.expect_contents(expected_script)
+        });
+}
```
Comment (jkomyno) [inline schema-engine/sql-migration-tests/tests/migrations/cockroachdb.rs:518]: Note to the reviewer: although I expected this test to fail against `main` when run on CockroachDB 23.1, it doesn't :/

This implies that somewhere in here we're not testing the "bad case for CockroachDB 22.2+" properly.

**Review by Jolg42 — APPROVED** [review]
Looks good to me

for info, I added a new build in our docker image `prismagraphql/cockroachdb-custom` for `23.1.13`
https://hub.docker.com/r/prismagraphql/cockroachdb-custom/tags

That updated `23.1` as well, which should in theory be `23.1.13` too now.

**Comment by Jolg42** [comment]
`SE: integration tests / cockroach_23_1 (pull_request)`
https://github.com/prisma/prisma-engines/actions/runs/7487743713/job/20380785635?pr=4632#step:7:2034

Looks like it's now failing like in 
chore(deps): update prismagraphql/cockroachdb-custom:23.1 docker digest to b6cbbf9 #4504
https://github.com/prisma/prisma-engines/pull/4504

**Comment by jkomyno** [comment]
> `SE: integration tests / cockroach_23_1 (pull_request)` https://github.com/prisma/prisma-engines/actions/runs/7487743713/job/20380785635?pr=4632#step:7:2034
> 
> Looks like it's now failing like in chore(deps): update prismagraphql/cockroachdb-custom:23.1 docker digest to b6cbbf9 #4504 #4504

I see the failing tests having a `maxValue` attribute in `sequence` as a test diff.
That comes out of https://github.com/prisma/prisma-engines/blob/0de06ae93d1d4ec6268874052d9c303e539e72be/schema-engine/connectors/sql-schema-connector/src/introspection/rendering/defaults.rs.

It's currently unclear to me why updating a patch version of CockroachDB would result in such a disruptive change.

**Comment by jkomyno** [comment]
I will revert https://github.com/prisma/prisma-engines/pull/4632/commits/b027610266d5d3642270fbd12cc4d20da7944f1e for now, and modify + review #4504 as part of Tech Debt Friday.

**Comment by Jolg42** [comment]
Indeed, a revert sounds good as we can tackle this separately.

**Comment by jkomyno** [comment]
(Seeing as you're already assigned to #4504, @Jolg42, we can also pair together on it!)

**Pull request merged**